src/AppInstallerCommonCore/Runtime.cpp

```diff
@@ -542,6 +540,11 @@ namespace AppInstaller::Runtime
         case PathName::PortableLinksUserLocation:
         case PathName::PortableLinksMachineLocation:
             result = GetPathDetailsCommon(path);
+            if (forDisplay)

```

Comment:
Sorry I may not make my previous comment clear enough. I was thinking to always call replace once. 

Is RoamingAppData used in these paths? I did not find the usage in this file. If not used, we should remove the RoamingAppData replace attempt.

If there are places using RoamingAppData, the paths should be broken down to 2 groups to avoid unnecessary replace prefix calls. Like:

case PathName::1:
case PathName::2:
case PathName::3:
            result = GetPathDetailsCommon(path);
            if (forDisplay)
            {
                ReplaceCommonPathPrefix(result.Path, GetKnownFolderPath(FOLDERID_LocalAppData), "%LOCALAPPDATA%");
            }
            break;
case PathName::4:
case PathName::5:
case PathName::6:
             result = GetPathDetailsCommon(path);
            if (forDisplay)
            {
                ReplaceCommonPathPrefix(result.Path, GetKnownFolderPath(FOLDERID_RoamingAppData), "%APPDATA%");
            }
            break;

Reply:
Winget doesn’t use roaming appdata by default, but since it is a runtime path, it's possible a user has changed the default to be in the roaming appdata folder, so I was trying to cover that possibility

Reply:
Yes, I think the roaming replace attempt should be removed. We are just trying to replace the value returned from one line above GetPathDetailsCommon(), where only LocalAppData is used. If user configures to use other paths in settings, it can be any path, we'll not be guessing their settings.

By the way, after looking at the code a bit closer, I think we should just try the LocalAppData replace with PortableLinksUserLocation and PortablePackageUserRoot only, for others, the replace seems very unlikely.

Reply:
Will do